src/Cms/App.php

```diff
@@ -1219,7 +1222,11 @@ public function path(): string
             return $this->path;
         }
 
-        return $this->path ??= $this->environment()->requestPath();
+        $current = $this->request()->path()->toString();
+        $index   = $this->environment()->baseUri()->path()->toString();
+        $path    = Str::afterStart($current, $index);
```

Comment:
This could fail if for some reason `$current` does not start with `$index`. In theory this should never happen, but if it does, this method will return a full URI including scheme and host. Maybe something like this:

```suggestion

        // catch edge cases that shouldn't occur
        // @codeCoverageIgnoreStart
        if (Str::startsWith($current, $index) !== true) {
            throw new LogicException('Could not determine request path');
        }
        // @codeCoverageIgnoreEnd
        
        $path = Str::afterStart($current, $index);
```

Reply:
I wouldn't do that to be honest. This is only about the path. Let's say for some weird reason the base URL has a different path than the request URL, then the request path would still be returned entirely, which sounds right to me.

Reply:
Right!